src/backend/src/services/slack.services.ts

```diff
@@ -0,0 +1,289 @@
+    //get the name of the channel from the slack api
+    try {
+      slackChannelName = await getChannelName(event.channel);
+    } catch (error) {
+      slackChannelName = `Unknown_Channel:${event.channel}`;
+    }
+    const dateCreated = new Date(Number(event.event_ts));
+
+    //get the message that will be processed either as the event or within a subtype
+    let eventMessage: SlackMessage;
+
+    if (event.subtype) {
+      switch (event.subtype) {
+        case 'message_deleted':
+          //delete the message using the client_msg_id
+          eventMessage = (event as SlackDeletedMessage).previous_message;
+          try {
+            return AnnouncementService.deleteAnnouncement(eventMessage.client_msg_id, organizationId);
+          } catch (ignored) {
+            return;
+          }
+        case 'message_changed':
+          eventMessage = (event as SlackUpdatedMessage).message;
+          break;
+        default:
+          //other events that do not effect announcements
+          return;
+      }
+    } else {
+      eventMessage = event as SlackMessage;
+    }
+
+    //loop through the blocks of the meta data while accumulating the
+    //text and users notified
+    let messageText = '';
+    let userIdsToNotify: string[] = [];
+
+    //Get the settings of all users in this organization to compare slack ids
+    const users = await UsersService.getAllUsers();
+    const userSettings = await Promise.all(
+      users.map((user) => {
+        return UsersService.getUserSettings(user.userId);
+      })
+    );
+
+    //get the name of the user that sent the message from slack
+    let userName: string = '';
+    try {
+      userName = (await getUserName(eventMessage.user)) ?? '';
+    } catch (ignored) {}
+
+    //if slack could not produce the name of the user, look for their name in prisma
+    if (!userName) {
+      const userIdList = userSettings
+        .filter((userSetting) => userSetting.slackId === eventMessage.user)
+        .map((userSettings) => userSettings.userId);
+      if (userIdList.length !== 0) {
+        const prismaUserName = users.find((user) => user.userId === userIdList[0]);
+        userName = prismaUserName
+          ? `${prismaUserName?.firstName} ${prismaUserName?.lastName}`
+          : 'Unknown User:' + eventMessage.user;
+      } else {
+        userName = 'Unknown_User:' + eventMessage.user;
+      }
+    }
+
+    //pull out the blocks of data from the metadata within the message event
+    const richTextBlocks = eventMessage.blocks?.filter((eventBlock: any) => eventBlock.type === 'rich_text');
+
+    if (richTextBlocks && richTextBlocks.length === 1) {
+      for (const element of richTextBlocks[0].elements[0].elements) {
+        messageText += await slackServices.blockToString(element);
+        userIdsToNotify = userIdsToNotify.concat(
+          await slackServices.blockToMentionedUsers(element, userSettings, event.channel)
+        );
+      }
+    } else {
+      return;
```

Comment:
add comment saying were disregarding messages without any mentions

Reply:
This else would indicate that the event we received does not line up with the documentation of the slack API and should never occur, even if no one is mentioned the event will still have those attributes

Reply:
what would the rich text block be then?

Reply:
The slack api documentation is kind of weird in that it doesn't do a great job of describing what attributes show up when, so in all the cases that I tested the rich_text block always showed up in the event, but the API documentation doesn't say that it is a required field, so this was my way of saying if it doesn't have that we will just ignore it.

Reply:
sounds good